### ISSUE TITLE
Incorrect output when two different lint libs being executed consecutively

### ISSUE THREAD

**Issue by EFanZh** [trailofbits/dylint#856]
Suppose I have two lint libraries in a workspace:

```toml
[workspace.metadata.dylint]
libraries = [{ path = "crates/my-lint-0" }, { path = "crates/my-lint-1" }]
```

If I run `cargo dylint` with these two different libraries consecutively, the second run will output the same result as the first one:

```sh
$ cargo dylint --lib my-lint-0
(some output)
$ cargo dylint --lib my-lint-1
(the same output from previous run)
```

I expect the second run should generate a different output.

**Comment by smoelius** [comment]
That sounds really annoying. I'll make fixing this a priority.

**Comment by smoelius** [comment]
Fixed by #866 and released in [2.4.3](https://github.com/trailofbits/dylint/releases/tag/v2.4.3). Please let me know if you have any problems.

Thanks a lot for reporting!
